client/app/index.js

```diff
@@ -14,6 +17,10 @@ const router = new VueRouter({ routes })
 
 // Initialize Application
 document.addEventListener('DOMContentLoaded', function initialize () {
+
+  Vue.component('cozy-dialog', Dialog)
```

Comment:
Please, avoid globally declared custom components, it makes the the code complex to understand when trying to know form where comes the components. Prefer the use of the `components` key inside the components that need them (see [local registration](https://vuejs.org/v2/guide/components.html#Local-Registration))